esphome/components/tuya/cover/tuya_cover.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/tuya/tuya.h"
+#include "esphome/components/cover/cover.h"
+
+namespace esphome {
+namespace tuya {
+
+class TuyaCover : public cover::Cover, public Component {
+ public:
+  void setup() override;
+  void dump_config() override;
+  void set_position_id(uint8_t dimmer_id) { this->position_id_ = dimmer_id; }
+  void set_tuya_parent(Tuya *parent) { this->parent_ = parent; }
+  void set_min_value(uint32_t min_value) { min_value_ = min_value; }
+  void set_max_value(uint32_t max_value) { max_value_ = max_value; }
+  void set_invert_position(bool invert_position) { invert_position_ = invert_position; }
+
+ protected:
+  void control(const cover::CoverCall &call) override;
+  cover::CoverTraits get_traits() override;
+
+  Tuya *parent_;
+  optional<uint8_t> position_id_{};
+  uint32_t min_value_ = 0;
+  uint32_t max_value_ = 100;
+  uint32_t value_range_;
+  bool invert_position_ = false;
```

Comment:
Just a style thing, we generally don't intiailize members by assignment:
```suggestion
  uint32_t min_value_{0};
  uint32_t max_value_{100};
  uint32_t value_range_;
  bool invert_position_{false};
```

Reply:
I am fine with either. `=` is more familiar to a lot of people =)

Reply:
According to the guidelines on esphome.io, initialization should not happen in code. All default values should only be part of the python configuration schema...

Reply:
#2415 =)